lib/active_merchant/billing/gateways/banwire.rb

```diff
@@ -32,61 +54,117 @@ def add_response_type(post)
         post[:response_format] = "JSON"
       end
 
-      def add_customer_data(post, options)
-        post[:user] = @options[:login]
-        post[:phone] = options[:billing_address][:phone]
-        post[:mail] = options[:email]
+      def add_auth_data(post, options)
+        post[:user] = @options[:login] if @options[:login] #- User (required) [alphanumeric] (Test: userdemo)
       end
 
-      def add_order_data(post, options)
-        post[:reference] = options[:order_id]
-        post[:concept] = options[:description]
+      def add_order_data(post, money, options)
+        post[:ORD_CURR_CD] = (options[:currency] || currency(money)).upcase #- Currency (optional) [text - length: 3] (Default: MXN)
+        post[:ORD_ID] = options[:order_id] #- Unique transaction ID (required) [alphanumeric - length: 1-22] (Example: 12345ABC)
+        post[:ORD_AMT] = amount(money) #- Transaction amount (required) [float - length: 10,2] (Example: 100.00)
+        post[:ORD_CONCEPT] = options[:description] #- Concept (required) [alphanumeric - lenght: 1-33] (Example: Compra de articulos)
       end
 
-      def add_address(post, creditcard, options)
-        post[:address] = options[:billing_address][:address1]
-        post[:post_code] = options[:billing_address][:zip]
+      def add_special_data(post, options)
+        post[:EBT_PREVCUST] = "N" #- Regular customer (required) [text - length: 1] (Y=Yes, N=No) (Default: N)
+        post[:EBT_DEVICEPRINT] = options[:device_print] if options[:device_print] #- Device print (optional) [text - lenght: 1-4000]
+        post[:CUST_IP] = options[:ip] if options[:ip] #- IP address of customer (required) [text - lenght: 255] (Example: 127.0.0.1)
+        post[:ebWEBSITE] = options[:web_site] if options[:web_site] #- URL website (required) [text - lenght: 1-60] (Example: www.comercio.com)
+        post[:PROD_DEL_CD] = options[:prod] || "DNP" #- Product type (required) [text - lenght: 3] [CNC=buy and ship, DCT=digital contents, DIG=digital goods, DNP=physical and digital, GFT=gift certificate, PHY=physical goods, REN=renewals and recharges, SVC=services] (Default: DNP) 
+        post[:NOTIFY_URL] = options[:notify_url] if options[:notify_url] #- Notify URL (required) [text - lenght: 255] (Example: www.comercio.com/response.php)
```

Comment:
Please default if required by API but not supplied. This is not an AM standard option, so will rarely be passed.